### PR TITLE
Story 15 - Fixed support for unreleased datasets

### PR DESCRIPTION
**Pull request by AndreasJJ**
It is now allowed to not provide an url on an unreleased dataset.

### CODE REVIEW

File: backend/Controllers/MetadataController.cs [hunk 1]
```diff
@@ -114,6 +114,9 @@ public async Task<IActionResult> PostAsync([FromBody] SaveMetadataResource resou
 			if (!ModelState.IsValid || !Enum.IsDefined(typeof(EReleaseState), resource.ReleaseState))
 				return BadRequest(ModelState.GetErrorMessages());
 
+			if (resource.ReleaseState != EReleaseState.Red && string.IsNullOrEmpty(resource.Url))
```
Comment (Letaguator) [inline backend/Controllers/MetadataController.cs]: Is this supposed to be:
```
if(resource.ReleaseState == EReleaseState.Released && string.IsNullOrEmpty(resource.Url)
```
?

Comment (AndreasJJ) [reply backend/Controllers/MetadataController.cs]: That depends on where we want to require an url.  As it is in the code now I only allow  unreleasable datasets to not have an url.

Comment (Letaguator) [reply backend/Controllers/MetadataController.cs]:
```C#
public enum EReleaseState: byte
	{
		[Description("The data is open")]
		Released = 1,

		[Description("Ready for release")]
		Green = 2,

		[Description("Needs work")]
		Yellow = 3,

		[Description("Not possible to release")]
		Red = 4,
	}
```

Comment (Letaguator) [reply backend/Controllers/MetadataController.cs]: That is the enum, I interpret it has Released is the only that requires url?

Comment (AndreasJJ) [reply backend/Controllers/MetadataController.cs]: I know how the enum looks like. But as i said it depends on where we want to require an url. If we want only released to require an url then the frontend needs to be changed too.

Comment (Letaguator) [reply backend/Controllers/MetadataController.cs]: @petterroea

File: backend/Controllers/MetadataController.cs [hunk 2]
```diff
@@ -114,6 +114,9 @@ public async Task<IActionResult> PostAsync([FromBody] SaveMetadataResource resou
 			if (!ModelState.IsValid || !Enum.IsDefined(typeof(EReleaseState), resource.ReleaseState))
 				return BadRequest(ModelState.GetErrorMessages());
 
+			if (resource.ReleaseState == EReleaseState.Released && string.IsNullOrEmpty(resource.Url))
+				return BadRequest("Url has to be supplied when releasestate is not unreleased (red)");
```
Comment (Letaguator) [inline backend/Controllers/MetadataController.cs]: Should the BadRequest information be updated too?

File: frontend/src/pages/SendMetadata/MetadataForm.jsx [hunk 3]
```diff
@@ -105,11 +138,19 @@ export const MetadataForm = () => {
 
   const handleSubmit = async (event) => {
     event.preventDefault();
+    setLoading(true);
     const { token } = userSelector.user;
+
+    const newState = { ...state };
+
+    if (newState.releaseState === 4 && (!newState.url || newState.url === '')) {
```
Comment (Letaguator) [inline frontend/src/pages/SendMetadata/MetadataForm.jsx]: What does this do? Does it set the url to null if releaseState is 4? Maybe it should do it if releaseState != 1?

Comment (AndreasJJ) [reply frontend/src/pages/SendMetadata/MetadataForm.jsx]: I suppose  😩

**Pull request merged**